### PR TITLE
End point to update a added todo item

### PR DESCRIPTION
**Pull request by vinay-jtc**
### Update a TODO item:

URI: PUT /todos/:id
Request body: { title: title }
Response HTTP Status: 200
Response body: { id: id, title: title }

## Database schema changes

- None

### Automated test cases added

1. Return 200 if a todo exists and the updated title is valid and mongo id is valid.
2. Return 400 if title is empty.
2. Return 400 if id is invalid mongo id.
3. Return 404 if todo item is not exist.

### Manual test cases run

1. Run by postman

### CODE REVIEW

**Comment by coder1033** [comment]
Please recheck your manual tests description.

File: src/validators/update-todo-item-validator.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { param, ValidationChain } from 'express-validator'
+import { AppContext } from '@typings'
+
+const updateTodoItemValidator = (appContext: AppContext): ValidationChain[] => [
+  param('id', 'VALIDATION_ERRORS.INVALID_ID').isMongoId(),
+]
+
+export default updateTodoItemValidator
```
Comment (coder1033) [inline src/validators/update-todo-item-validator.ts]: Add a new line

File: src/validators/update-todo-item-validator.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { param, ValidationChain } from 'express-validator'
+import { AppContext } from '@typings'
+
+const updateTodoItemValidator = (appContext: AppContext): ValidationChain[] => [
+  param('id', 'VALIDATION_ERRORS.INVALID_ID').isMongoId(),
```
Comment (coder1033) [inline src/validators/update-todo-item-validator.ts:5]: Also validate the title in this validator itself.

File: test/spec/todo/todo.spec.ts [hunk 3]
```diff
@@ -46,31 +46,49 @@ describe("POST /todos", () => {
   });
 });
 
-describe("GET /todos/:id", () => {
-  it("should fetch a todo item if it exists and if id is valid mongo id", async () => {
-    const todoItem = await testAppContext.TodoItemRepository.save(
-      new TodoItem({ title: "Fetching an item" })
+describe("PUT /todos/:id", () => {
+  it("should return 200 if todo exists and title is validately true and 400 if title is empty or not a string else 404", async () => {
```
Comment (coder1033) [inline test/spec/todo/todo.spec.ts]: Please make separate test block for each case.
Do not mix multiple tests in one `it` block.

File: src/validators/update-todo-item-validator.ts (hunk 2 above)
Comment (vinay-jtc) [reply src/validators/update-todo-item-validator.ts:5]: updated

File: src/validators/update-todo-item-validator.ts (hunk 1 above)
Comment (vinay-jtc) [reply src/validators/update-todo-item-validator.ts]: updated

File: test/spec/todo/todo.spec.ts (hunk 3 above)
Comment (vinay-jtc) [reply test/spec/todo/todo.spec.ts]: Updated

File: test/spec/todo/todo.spec.ts [hunk 4]
```diff
@@ -45,22 +46,58 @@ describe('POST /todos', () => {
...
-    expect(res.body).to.be.an('array');
-  });
+    expect(res.body).to.have.property('id');
+    expect(res.body).to.have.property('title');
+  })
 
-  it('we should check if the array returned is empty when there are no todo items', async () => {
-    await testAppContext.TodoItemRepository.getAll();
+  it('should return 400 if todo exists & id is valid mongo id & title is empty', async () => {
+    let todoItem = await testAppContext.TodoItemRepository.save(
```
Comment (coder1033) [inline test/spec/todo/todo.spec.ts]: `const`

File: test/spec/todo/todo.spec.ts [hunk 5]
```diff
@@ -45,22 +46,58 @@ describe('POST /todos', () => {
...
+        title: '',
+      });
+    expect(res).to.have.status(400);
+    expect(res.body)
+      .to.have.nested.property('failures[0].message')
+      .to.equal('Please specify the valid title');
+  })
 
-    await testAppContext.TodoItemRepository.deleteMany({});
+  it('should return 400 if id is valid mongo id', async () => {
```
Comment (coder1033) [inline test/spec/todo/todo.spec.ts]: `invalid`

**Comment by coder1033** [comment]
You have written 4 tests. Describe them individually in PR description.

**Review by coder1033 — APPROVED** [review]

File: test/spec/todo/todo.spec.ts (hunk 5 above)
Comment (vinay-jtc) [reply test/spec/todo/todo.spec.ts]: Updated

File: test/spec/todo/todo.spec.ts (hunk 4 above)
Comment (vinay-jtc) [reply test/spec/todo/todo.spec.ts]: Updated

**Comment by vinay-jtc** [comment]
Described all 4 test cases and going to merge.

**Pull request merged**